app/models/speaker.rb

```diff
@@ -9,8 +9,15 @@ class Speaker < ActiveRecord::Base
 
   serialize :info, Hash
 
-  validates :user, :event, presence: true
+  validates :event, presence: true
   validates :bio, length: {maximum: 500}
+  validates :name, :email, presence: true
+
+  scope :with_program_session, -> { Speaker.where("program_session_id IS NOT NULL") }
+
+  scope :for_event, -> (current_event) do
```

Comment:
I wonder if we can write better looking code with a has_many on event vs using this type of scope.